pipelines/deployer/deployer.yaml

```
@@ -639,6 +639,39 @@ jobs:
     config: *unpack_release
     params:
       CLUSTER_PUBLIC_KEY: ((ci-system-gpg-public))
+  - task: scale-autoscaler-down
+    image: task-toolbox
+    timeout: 10s
+    params:
+      CLUSTER_NAME: ((cluster-name))
+      ACCOUNT_ROLE_ARN: ((account-role-arn))
+    config:
+      platform: linux
+      run:
+        path: /bin/bash
+        args:
+        - -euo
+        - pipefail
+        - -c
+        - |
+          echo "assuming aws deployer role..."
+          AWS_CREDS="$(aws-assume-role $ACCOUNT_ROLE_ARN)"
+          echo "Looking for existing EKS cluster"
+          CLUSTER_COUNT=$(aws eks list-clusters | jq '.clusters[] | select (. == "$CLUSTER_NAME") | 1' | wc -l)
+          if [ "$CLUSTER_COUNT" == "1" ]; then
+              echo "found it, fetching kubeconfig from aws..."
+              aws eks update-kubeconfig \
+                --name "${CLUSTER_NAME}" \
+                --kubeconfig ./kubeconfig
+              echo "done, looking for deployments"
+              export KUBECONFIG=$(pwd)/kubeconfig
+              DEPLOYMENT_COUNT=$(kubectl --namespace gsp-system get deployment -o json | jq '.items[] | select (.metadata.name == "gsp-aws-cluster-autoscaler") | 1' | wc -l)
+              if [ "$DEPLOYMENT_COUNT" == "1" ]; then
+                  echo "found it, scaling down to 0 replicas"
+                  kubectl --namespace gsp-system scale --replicas=0 deployment/gsp-aws-cluster-autoscaler
```

as an alternative to the above, why bother checking if the deployment is there? Why not do something like

```suggestion
                  kubectl --namespace gsp-system scale --replicas=0 deployment/gsp-aws-cluster-autoscaler || true
```

and remove the above `kubectl get deployment` stuff?

---

We still want it to error if kubectl fails to contact the cluster for whatever reason

---

Although actually maybe we don't as the terraform change we'd be blocking the application of may be fixing that. Hmm.